nessie/s3_gateway_test.go

```diff
@@ -104,3 +106,133 @@ func TestS3UploadAndDownload(t *testing.T) {
 		})
 	}
 }
+
+func TestS3CopyObject(t *testing.T) {
+	ctx, _, repo := setupTest(t)
+	destRepo := createRepositoryByName(ctx, t, "tests3copyobjectdest")
+
+	accessKeyID := viper.GetString("access_key_id")
+	secretAccessKey := viper.GetString("secret_access_key")
+	endpoint := viper.GetString("s3_endpoint")
+	opts := minio.PutObjectOptions{}
+	rand := rand.New(rand.NewSource(17))
+
+	creds := sigs[0].GetCredentials(accessKeyID, secretAccessKey, "")
+
+	type Object struct {
+		SourcePath, DestPath, Content string
+	}
+
+	minioClient, err := minio.New(endpoint, &minio.Options{
+		Creds:  creds,
+		Secure: false,
+	})
+	if err != nil {
+		t.Fatalf("minio.New: %s", err)
+	}
+
+	o := Object{
+		Content:    testutil.RandomString(rand, randomDataContentLength),
+		SourcePath: prefix + "source-file",
+		DestPath:   prefix + "dest-file",
+	}
```

Comment:
```suggestion

	Content:=    testutil.RandomString(rand, randomDataContentLength)
	SourcePath:= prefix + "source-file"
	DestPath: =  prefix + "dest-file"

```

Reply:
You don't really need the Object type, also it's a bit confusing that the Object has two paths